src/main/java/org/jahia/modules/sitemap/listeners/SitemapSiteListener.java

```diff
@@ -70,10 +70,30 @@ public void onEvent(EventIterator events) {
                 }
                 // Site node removed
                 if (event.getType() == Event.NODE_REMOVED && event.getPath().equals("/sites/" + siteKey)) {
+
                     SitemapJobBuilder builder = new SitemapJobBuilder();
                     builder.createJob = false;
                     jobsToBuild.put(siteKey, builder);
                 }
+                if (event.getType() == Event.PROPERTY_CHANGED && event.getPath().equals("/sites/" + siteKey + "/j:installedModules")) {
+
+                    JCRPropertyWrapper prop = (JCRPropertyWrapper) session.getItem(eventPath);
+                   boolean siteMapNotInstalled = Arrays.stream(prop.getValues()).noneMatch(value -> {
```

Comment:
This can be wrapped with a `ThrowingPredicate.unchecked` instead of the try/catch block